src/Middleware/OutputCaching/src/Streams/SegmentWriteStream.cs

```diff
@@ -125,7 +125,7 @@ public override void Write(byte[] buffer, int offset, int count)
         ValidateBufferArguments(buffer, offset, count);
         if (!CanWrite)
         {
-            throw new InvalidOperationException("The stream has been closed for writing.");
+            throw new ObjectDisposedException(nameof(SegmentWriteStream), "The stream has been closed for writing.");
```

Comment:
why ODE?

Reply:
Because it's ObjectDisposedException today - https://github.com/dotnet/aspnetcore/blob/cefc6ccf4b23ac9f07418cd10e374cf533276eee/src/Middleware/OutputCaching/src/Streams/SegmentWriteStream.cs#L128

Reply:
ah, so this was just an intermediate commit changing them back to that? got it